### PR TITLE
Currency-Fragments:

### LINKED ISSUE #20: Migrate currency from activity to fragment
BottomNavigationView requires view to be a Fragment. Now currency screen is designed as an activity. It needs to be changed to a Fragment. There must be little to no problems implementing it, but it needs to be done, anyway. For any reference needed ask me (@kramarov-evg) or Google

### PR DESCRIPTION
**Pull request by LeonidVolohov**
Migrate currency from activity to fragment #20

### CODE REVIEW

File: app/src/main/java/com/stonks/api/ApiService.kt [hunk 1]
```diff
@@ -2,22 +2,31 @@ package com.stonks.api
...
-    .baseUrl(CURRENCY_API_BASE_URL)
-    .addConverterFactory(GsonConverterFactory.create())
-    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
-    .build()
-    .create(CurrencyRequests::class.java)
+        .baseUrl(CURRENCY_API_BASE_URL)
+        .addConverterFactory(GsonConverterFactory.create())
+        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
+        .build()
+        .create(CurrencyRequests::class.java)
```
Comment (kramarov-evg) [inline app/src/main/java/com/stonks/api/ApiService.kt]: Why extra-indented? Code-style

File: app/src/main/java/com/stonks/api/ApiService.kt [hunk 2]
```diff
@@ -2,22 +2,31 @@ package com.stonks.api
...
+        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
+        .build()
+        .create(CryptoCurrencyRequests::class.java)
+
+val stocksApi = Retrofit.Builder()
+        .baseUrl(STOCK_API_BASE_URL)
+        .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
+        .addConverterFactory(GsonConverterFactory.create())
+        .build()
+        .create(StocksRequests::class.java)
```
Comment (kramarov-evg) [inline app/src/main/java/com/stonks/api/ApiService.kt]: Same about extra-indentation here

File: app/src/main/java/com/stonks/api/Constants.kt [hunk 3]
```diff
@@ -4,8 +4,9 @@ class Constants {
     companion object {
         const val CURRENCY_API_BASE_URL = "https://api.exchangeratesapi.io/"
 
-        const val STOCK_API_BASE_URL = "https://www.alphavantage.co/"
+        const val STOCK_API_BASE_URL = "https://www.alphavantage.co/query/"
```
Comment (kramarov-evg) [inline app/src/main/java/com/stonks/api/Constants.kt]: No need to add "query" endpoint. All requests are anyway sent there via `@GET("endpoint")`, we've discussed it already and found it useless

File: app/src/main/java/com/stonks/api/stocks/StocksApiDataUtils.kt [hunk 4]
```diff
@@ -0,0 +1,187 @@
+package com.stonks.api.stocks
```
Comment (kramarov-evg) [inline app/src/main/java/com/stonks/api/stocks/StocksApiDataUtils.kt]: Why are you even changing anything related to stocks? Those are my files!

File: app/src/main/java/com/stonks/api/stocks/StocksDataModel.kt [hunk 5]
```diff
@@ -0,0 +1,38 @@
+package com.stonks.api.stocks
```
Comment (kramarov-evg) [inline app/src/main/java/com/stonks/api/stocks/StocksDataModel.kt]: DON'T TOUCH STOCKS!!!

File: app/src/main/java/com/stonks/api/stocks/StocksRequests.kt [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package com.stonks.api.stocks
```
Comment (kramarov-evg) [inline app/src/main/java/com/stonks/api/stocks/StocksRequests.kt]: STOCKS. ARE. MINE.

File: app/src/main/java/com/stonks/ui/chart/StockLineChart.kt [hunk 7]
```diff
@@ -40,7 +40,7 @@ class StockLineChart(lineChart: LineChart) {
         lineChart.xAxis.labelCount = 4
     }
 
-    fun getLineData(entries: List<String>, baseCurrency: String, targetCurrency: String): LineDataSet {
+    fun getLineData(entries: List<Double>, baseCurrency: String, targetCurrency: String): LineDataSet {
```
Comment (kramarov-evg) [inline app/src/main/java/com/stonks/ui/chart/StockLineChart.kt:43]: Thanks here :+1:

File: app/src/main/java/com/stonks/ui/currency/CurrencyFragment.kt [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+            if (isChecked) {
+                when (currency_button_group.checkedButtonId) {
+                    R.id.currency_togglebutton_one_week_selector -> {
+                        currencyFragmentUtils.plotRatesPerPeriod(
+                                startDate = "1W",
+                                targetRate = targetRateSpinnerString,
+                                baseRate = baseRateSpinnerString,
+                                stockLineChart = currencyLineChart,
+                                currencyChart = currency_chart
+                        )
```
Comment (kramarov-evg) [inline app/src/main/java/com/stonks/ui/currency/CurrencyFragment.kt:93]: Add 1D button back. We MUST have it according to TZ. We just can say, that we have no data for the period selected (like a toast, or any other way), but we MUST provide an opportunity to select 1 day

File: app/src/main/java/com/stonks/ui/currency/CurrencyFragmentUtils.kt [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+const val DEFAULT_CURRENCY_INDEX = 17
+const val PRIMARY_RATES = "USD,EUR,GBP,JPY,CHF"
+
+class CurrencyFragmentUtils(disposable: Disposable?) {
+    private val TAG = CurrencyFragmentUtils::class.java.name
+    private var localDisposable = disposable
+
+    /**
+     * Умножает две строки [firstNumber]  и [secondNumber] и возвращает только первые 7 символов
+     */
```
Comment (kramarov-evg) [inline app/src/main/java/com/stonks/ui/currency/CurrencyFragmentUtils.kt:23]: If you're documenting functions it's a good idea to specify what are their params and return values. Not really a deadly sin here, but just for future & to refactor later

File: app/src/main/java/com/stonks/ui/main/MainActivity.kt [hunk 10]
```diff
@@ -37,7 +41,4 @@ class MainActivity : AppCompatActivity() {
 
             true
         }
-
-    // Variable storing currently selected fragment for displaying
-    private var selectedFragment : Fragment = DefaultFragment()  // TODO: place fragment corresponding to most left tab (now crypto)
```
Comment (kramarov-evg) [inline app/src/main/java/com/stonks/ui/main/MainActivity.kt:42]: Why did you delete this and not replaced `DefaultFragment()` with `CurrencyFragment()`?

File: app/src/main/java/com/stonks/ui/stocks/StocksFragment.kt [hunk 11]
```diff
@@ -1,14 +1,165 @@
 package com.stonks.ui.stocks
```
Comment (kramarov-evg) [inline app/src/main/java/com/stonks/ui/stocks/StocksFragment.kt]: Keep stocks out of your commits!!!

**Review by kramarov-evg — CHANGES_REQUESTED** [review]
In general nice, but some things have to be fixed. Also some for-future comments. See it

File: app/src/main/java/com/stonks/api/ApiService.kt (hunk 1 above)
Comment (LeonidVolohov) [reply app/src/main/java/com/stonks/api/ApiService.kt]: Android Studio does it automatically when "Reformat code" selected during commits

File: app/src/main/java/com/stonks/api/ApiService.kt (hunk 2 above)
Comment (LeonidVolohov) [reply app/src/main/java/com/stonks/api/ApiService.kt]: Wrote the message above about it

File: app/src/main/java/com/stonks/ui/currency/CurrencyFragmentUtils.kt (hunk 9 above)
Comment (LeonidVolohov) [reply app/src/main/java/com/stonks/ui/currency/CurrencyFragmentUtils.kt:23]: According to [this](https://stackoverflow.com/questions/38015307/generating-kotlin-method-class-comments/51067319#51067319) SO answer it is better to comment with this style

File: app/src/main/java/com/stonks/ui/main/MainActivity.kt (hunk 10 above)
Comment (LeonidVolohov) [reply app/src/main/java/com/stonks/ui/main/MainActivity.kt:42]: It is not deleted just replaced to the top of the class

File: app/src/main/java/com/stonks/ui/currency/CurrencyFragment.kt (hunk 8 above)
Comment (LeonidVolohov) [reply app/src/main/java/com/stonks/ui/currency/CurrencyFragment.kt:93]: Added

**Review by kramarov-evg — APPROVED** [review]

**Review by owlerxan — APPROVED** [review]
